addons/hearing/functions/fnc_firedNear.sqf

```diff
@@ -47,10 +47,18 @@ if (_silencer != "") then {
     _audibleFireCoef = getNumber (configFile >> "CfgWeapons" >> _silencer >> "ItemInfo" >> "AmmoCoef" >> "audibleFire");
 };
 
-_audibleFire = getNumber (configFile >> "CfgAmmo" >> _ammo >> "audibleFire");
-
-_loudness = _audibleFireCoef * _audibleFire / 64;
-_strength = _vehAttenuation * (_loudness - (_loudness/50 * _distance)); // linear drop off
+_magazine = (getArray(configFile >> "CfgWeapons" >> _weapon >> "magazines")) select 0;
+_initSpeed = getNumber(configFile >> "CfgMagazines" >> _magazine >> "initSpeed");
+_ammoConfig = (configFile >> "CfgAmmo" >> _ammo);
+_caliber = getNumber(_ammoConfig >> "ACE_caliber");
+_parentClasses = [_ammoConfig, true] call BIS_fnc_returnParents;
+if ("RocketBase" in _parentClasses || "MissileBase" in _parentClasses) then {
+    _caliber = 200;
+} else {
+    if (_caliber <= 0) then { _caliber = 6.5; };
+};
+_loudness = (_caliber ^ 1.25 / 10) * (_initspeed / 1000) * _audibleFireCoef / 5;
```

Comment:
loudness also needs to take into account the presence of supressors/silencers, which are attachments in A3.